pkgs/applications/networking/browsers/firefox/wrapper.nix

```diff
@@ -324,6 +324,8 @@ let
             --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH" \
             --suffix XDG_DATA_DIRS : '${gnome.adwaita-icon-theme}/share' \
             --set-default MOZ_ENABLE_WAYLAND 1 \
+            ${lib.optionalString (nativeMessagingHosts != []) "--run \"mkdir -p ~/.mozilla/native-messaging-hosts\""} \
+            ${lib.concatMapStringsSep " " (ext: "--run \"ln -sfLt ~/.mozilla/native-messaging-hosts ${ext}/lib/mozilla/native-messaging-hosts/*\"") nativeMessagingHosts} \
```

Comment:
```suggestion
            ${lib.concatMapStringsSep " " (ext: "--run \"lndir ~/.mozilla/native-messaging-hosts ${ext}/lib/mozilla/native-messaging-hosts/*\"") nativeMessagingHosts} \
```

Reply:
lndir is by default not in PATH and if we use that, we need to properly clean up the directories to not leave stale files behind.